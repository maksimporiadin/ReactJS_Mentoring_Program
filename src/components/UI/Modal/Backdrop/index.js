import React from 'react';
import './style.css';

const Backdrop = ({ show, close }) => (
    show ? <div className="backdrop" onClick={close}></div> : null
);

export default Backdrop;