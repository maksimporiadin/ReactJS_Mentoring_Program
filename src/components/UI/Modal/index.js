import React, { Component } from 'react';

import './style.css';
import Backdrop from './Backdrop';

class Modal extends Component {
    shouldComponentUpdate(nextProps, nextState) {
        return nextProps.show !== this.props.show || nextProps.children !== this.props.children;
    }

    render () {
        return (
            <React.Fragment>
                <Backdrop show={this.props.show} close={this.props.closeModal}/>
                <div
                    className="modal-custom"
                    onClick={this.props.closeModal}
                    style={{
                         transform: this.props.show ? 'translateY(0)' : 'translateY(-100vh)',
                         opacity: this.props.show ? '1' : '0'
                     }}>
                    {this.props.children}
                </div>
            </React.Fragment>
        )
    }
}

export default Modal;
