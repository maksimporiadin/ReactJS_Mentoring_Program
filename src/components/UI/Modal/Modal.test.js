import 'jsdom-global/register';
import React from 'react';
import Modal  from './index';
import { shallow } from 'enzyme';

describe('Modal', () => {
    let wrapper;
    const defaultProps = {
        closeModal() {},
    };

    beforeEach(()=>{
        jest.resetAllMocks();
        wrapper = shallow(<Modal />);
    });

    it('is renders with show: true', () => {
        const props = {
            show: true
        };
        wrapper = shallow(<Modal {...defaultProps} {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('is renders with show: false', () => {
        const props = {
            show: false
        };
        wrapper = shallow(<Modal {...defaultProps} {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });
});