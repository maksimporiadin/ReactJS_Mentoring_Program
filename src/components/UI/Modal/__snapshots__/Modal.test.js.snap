// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal is renders with show: false 1`] = `
<Fragment>
  <Backdrop
    close={[Function]}
    show={false}
  />
  <div
    className="modal-custom"
    onClick={[Function]}
    style={
      Object {
        "opacity": "0",
        "transform": "translateY(-100vh)",
      }
    }
  />
</Fragment>
`;

exports[`Modal is renders with show: true 1`] = `
<Fragment>
  <Backdrop
    close={[Function]}
    show={true}
  />
  <div
    className="modal-custom"
    onClick={[Function]}
    style={
      Object {
        "opacity": "1",
        "transform": "translateY(0)",
      }
    }
  />
</Fragment>
`;
